components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/db/TabCollectionDao.kt

```diff
@@ -48,6 +48,7 @@ internal interface TabCollectionDao {
     )
     fun getTabCollections(): Flow<List<TabCollectionWithTabs>>
 
+    @Transaction
```

Comment:
Are we intentionally adding @Transaction for this query because the returned list could be fairly big and therefore want consistent result or for some other reason?

Reply:
This is probably for consistency. The Room upgrade must have added a new lint rule to ensure we were always adding the `@Transaction` annotation.

It probably came in from [here](https://developer.android.com/jetpack/androidx/releases/room#2.4.0-alpha03).

Reply:
Indeed, thanks Jonathan. 
After updating to a new `room` version we'd get a warning here saying 
> The return value includes a POJO with a @Relation. It is usually desired to annotate this method with @Transaction to avoid possibility of inconsistent results between the POJO and its relations.